src/libostree/ostree-sign-dummy.c

```diff
@@ -0,0 +1,205 @@
+
+  OstreeSignDummy *sign =  _ostree_sign_dummy_get_instance_private(OSTREE_SIGN_DUMMY(self));
+
+  if (sign->pk_ascii != NULL)
+    g_free(sign->pk_ascii);
+
+  sign->pk_ascii = g_variant_dup_string (key, 0);
+
+  return TRUE;
+}
+
+gboolean ostree_sign_dummy_data (OstreeSign *self,
+                                 GBytes *data,
+                                 GBytes **signature,
+                                 GCancellable *cancellable,
+                                 GError **error)
+{
+
+  g_debug ("%s enter", __FUNCTION__);
+  g_return_val_if_fail (OSTREE_IS_SIGN (self), FALSE);
+
+  OstreeSignDummy *sign =  _ostree_sign_dummy_get_instance_private(OSTREE_SIGN_DUMMY(self));
+
+  *signature = g_bytes_new (sign->sk_ascii, strlen(sign->sk_ascii));
+
+  return TRUE;
+}
+
+const gchar * ostree_sign_dummy_get_name (OstreeSign *self)
+{
+  g_debug ("%s enter", __FUNCTION__);
+  g_return_val_if_fail (OSTREE_IS_SIGN (self), FALSE);
+
+  return OSTREE_SIGN_DUMMY_NAME;
+}
+
+const gchar * ostree_sign_dummy_metadata_key (OstreeSign *self)
+{
+  g_debug ("%s enter", __FUNCTION__);
+
+  return OSTREE_SIGN_METADATA_DUMMY_KEY;
+}
+
+const gchar * ostree_sign_dummy_metadata_format (OstreeSign *self)
+{
+  g_debug ("%s enter", __FUNCTION__);
+
+  return OSTREE_SIGN_METADATA_DUMMY_TYPE;
+}
+
+gboolean ostree_sign_dummy_data_verify (OstreeSign *self,
+                                            GBytes     *data,
+                                            GVariant   *signatures,
+                                            GError     **error)
+{
+  g_debug ("%s enter", __FUNCTION__);
+  g_return_val_if_fail (OSTREE_IS_SIGN (self), FALSE);
+  g_return_val_if_fail (data != NULL, FALSE);
+
+  OstreeSignDummy *sign =  _ostree_sign_dummy_get_instance_private(OSTREE_SIGN_DUMMY(self));
+
+  gboolean ret = FALSE;
+
+  if (signatures == NULL)
+    {
+      g_set_error_literal (error,
+                           G_IO_ERROR, G_IO_ERROR_FAILED,
+                           "signature: dummy: commit have no signatures of my type");
+      goto err;
+    }
+
+
+  if (!g_variant_is_of_type (signatures, (GVariantType *) OSTREE_SIGN_METADATA_DUMMY_TYPE))
+    {
+      g_set_error_literal (error,
+                           G_IO_ERROR, G_IO_ERROR_FAILED,
+                           "signature: dummy: wrong type passed for verification");
+      goto err;
+    }
+
+  for (gsize i = 0; i < g_variant_n_children(signatures); i++)
+    {
+      g_autoptr (GVariant) child = g_variant_get_child_value (signatures, i);
+      g_autoptr (GBytes) signature = g_variant_get_data_as_bytes(child);
+
+      gsize sign_size = 0;
+      g_bytes_get_data (signature, &sign_size);
+      g_autofree gchar *sign_ascii = g_strndup(g_bytes_get_data (signature, NULL), sign_size);
+      g_debug("Read signature %d: %s", (gint)i, sign_ascii);
+      g_debug("Stored signature %d: %s", (gint)i, sign->pk_ascii);
+
+      if (!g_strcmp0(sign_ascii, sign->pk_ascii))
+          ret = TRUE;
```

Comment:
Why not just `return TRUE` here?

Reply:
my bad. Indeed, with a new style such kind of optimizations are more visible. Thanks!